src/main/java/htsjdk/tribble/AsciiFeatureCodec.java

```diff
@@ -49,14 +50,19 @@ public boolean isDone(final LineIterator lineIterator) {
     }
 
     @Override
-    public LocationAware makeIndexableSourceFromStream(final InputStream bufferedInputStream) {
-        final PositionalBufferedStream pbs;
-        if (bufferedInputStream instanceof PositionalBufferedStream) {
-            pbs = (PositionalBufferedStream) bufferedInputStream;
+    public LocationAware makeIndexableSourceFromStream(final InputStream inputStream) {
+        if (inputStream instanceof BlockCompressedInputStream) {
```

Comment:
I think that would put an unnecessary constraint on AsciiLineReader (it should be legal to wrap an AsciiLineReader around a BCS, though having said that, its confusing because AsciiLineReader confers both a line orientation as well as buffering, which is what messes up the positions). 

Also, given how brittle all this code is, I like the idea of having this behavior and the comments explicitly included right here, where the explicit control over the buffering/positioning matters (this method is specifically to support indexing).